### PR TITLE
Swagger doc

### PR DESCRIPTION
**Pull request by noctera**
| Status  | Type  | Env Vars Change |
| :---: | :---: | :---: |
| :x: Hold | Feature | No |

## Description

Swagger docs for api routes

run `npm run swagger` to see them

## Checklist

- [x] I have read the **CONTRIBUTING** document.
- [x] I have considered the accessibility of my changes (i.e. did I add proper content descriptions to images, or run my changes with talkback enabled?)
- [x] I have documented my code if needed

### CODE REVIEW

File: docs/api/swagger.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "info": {
     "title": "vocascan",
     "description": "A highly configurable vocabulary trainer.",
-    "version": "1.0.0"
+    "version": "v0.1-alpha"
```
Comment (luwol03) [inline docs/api/swagger.json]: ?

File: docs/api/swagger.json [hunk 2]
```diff
@@ -69,47 +75,927 @@
...
+      }
+    },
+    "/languagePackage": {
+      "post": {
+        "summary": "Create language package for user",
+        "tags": [
+          "language package"
+        ],
+        "operationId": "createLanguagePackage",
+        "security": [],
```
Comment (luwol03) [inline docs/api/swagger.json]: the `security: []` means no auth. But for all other fields than the login fields auth is required. So please remove it also on all others.

File: docs/api/swagger.json [hunk 3]
```diff
@@ -69,47 +75,927 @@
...
+        "schema": {
+          "type": "boolean"
+        }
+      },
+      "nativeNames": {
+        "in": "query",
+        "name": "nativeNames",
+        "description": "decides if nativeNames gets returned",
+        "schema": {
+          "type": "boolean"
```
Comment (luwol03) [inline docs/api/swagger.json]: Add a default values. Also in other parameters. See https://swagger.io/docs/specification/describing-parameters/ under `Default Parameter Values` section.

File: docs/api/swagger.json [hunk 4]
```diff
@@ -69,47 +75,927 @@
...
+            "$ref": "#/components/responses/404"
+          }
+        }
+      }
+    },
+    "/info": {
+      "get": {
+        "summary": "Get server info",
+        "tags": [
+          "info"
```
Comment (luwol03) [inline docs/api/swagger.json]: maybe you can put this into the misc tag. *(Dont forget to move this up to hold it cleanly)*

File: docs/api/swagger.json [hunk 5]
```diff
@@ -69,47 +75,927 @@
...
+          "name": {
+            "type": "string"
+          },
+          "nativeNames": {
+            "type": "array",
+            "items": {
+              "name": {
+                "type": "string"
+              }
+            }
```
Comment (luwol03) [inline docs/api/swagger.json]: array cannot have name property with object. Change to following to fix this error:
image

```suggestion
            "items": {
              "type": "string"
            }
```

**Review by luwol03 — CHANGES_REQUESTED** [review]

File: docs/api/swagger.json (hunk 1 above)
Comment (noctera) [reply docs/api/swagger.json]: I pasted some code, which i removed, but seems like I accidentally overlooked that I changed the version

Comment (noctera) [reply docs/api/swagger.json]: done

**Review by luwol03 — APPROVED** [review]

**Pull request merged**